### ISSUE TITLE
Avoid unnecessary implicit assignments in our own test suite

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1820]
Still not 100% sure this necessitates excepting `expect_warning()`

e.g. even in case the value needs to be re-used in a subsequent test, you can do
```r
# bad
expect_warning(prepared_thing <- my_setup_with_warning(), "setup warning")
expect_equal(funny_function(prepared_thing), funny_value) |>
  expect_warning("funny warning")

# good
my_setup_with_warning() |>
  funny_function() |>
  expect_equal(funny_value) |>
  expect_warning("setup warning") |>
  expect_warning("funny warning")
```

This is also how we structure our tests that expect warnings.

_Originally posted by @AshesITR in https://github.com/r-lib/lintr/issues/1777#issuecomment-1343414735_
